app/celery/tasks.py

```diff
@@ -213,24 +216,25 @@ def save_smss(self, service_id: Optional[str], signed_notifications: List[Any],
     is not None then it is passed to the RedisQueue to let it know it
     can delete the inflight notifications.
     """
-    verified_notifications: List[Any] = []
+    verified_notifications: List[VerifiedNotification] = []
     notification_id_queue: Dict = {}
-    saved_notifications = []
+    saved_notifications: List[Notification] = []
     for signed_notification in signed_notifications:
         try:
-            notification = signer.verify(signed_notification)
+            _notification: NotificationDictToSign = signer.verify(signed_notification)
         except BadSignature:
             current_app.logger.exception(f"Invalid signature for signed_notification {signed_notification}")
             raise
-        service_id = notification.get("service_id", service_id)  # take it it out of the notification if it's there
+        service_id = _notification.get("service_id", service_id)  # take it it out of the notification if it's there
         service = dao_fetch_service_by_id(service_id, use_cache=True)
 
         template = dao_get_template_by_id(
-            notification.get("template"), version=notification.get("template_version"), use_cache=True
+            _notification.get("template"), version=_notification.get("template_version"), use_cache=True
         )
-        sender_id = notification.get("sender_id")
-        notification_id = notification.get("id", create_uuid())
-        notification["notification_id"] = notification_id
+        # bug: _notification does not have "sender_id" key
```

Comment:
can you put it as a TODO - I think we should remove this in the next PR - but merge this first without removing it

Reply:
Yes good idea! I'll do that.